### ISSUE TITLE
`sidebar.hidden` doesn't work if the file is in the `docs/` directory

### ISSUE THREAD

**Issue by noClaps** [withastro/starlight#481]
### What version of `starlight` are you using?

0.7.0

### What version of `astro` are you using?

2.10.5

### What package manager are you using?

bun, pnpm (in example)

### What operating system are you using?

Mac

### What browser are you using?

Safari

### Describe the Bug

Setting the `sidebar.hidden` option on the `index.mdx` page doesn't hide the page from the sidebar. In my case, this occurs when page is at `src/content/docs/index.mdx`. It also doesn't hide any other page if the option is set on that page.

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-t7f89g?file=src%2Fcontent%2Fdocs%2Findex.mdx

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by HiDeoo** [comment]
Indeed, it looks like the [filter](https://github.com/withastro/starlight/blob/c04c79785f7ac343edc5565a5f1faed929fe350a/packages/starlight/utils/navigation.ts#L89-L90) is only [applied](https://github.com/withastro/starlight/blob/c04c79785f7ac343edc5565a5f1faed929fe350a/packages/starlight/utils/navigation.ts#L62-L63) when there is a sidebar group explicitly configured with `autogenerate` which is [not the case](https://github.com/withastro/starlight/blob/c04c79785f7ac343edc5565a5f1faed929fe350a/packages/starlight/utils/navigation.ts#L247-L250) when there is no sidebar config at all.

I guess technically the entries are "autogenerated" 🤔 Not sure if this is intended or just a missed edge case tho.
